doc/dev/crimson/poseidonstore.rst

```diff
@@ -0,0 +1,401 @@
+    +--| Entry0  | ...     | ...     |                          | <Offset, len> |       |          |
+    |  +---------+---------+---------+                          | <Offset, len> |-------|----+     |
+    |  | ...     | ...     | ...     |                          | ...           |       |    |     |
+    |  +---------+---------+---------+                          +---------------+       |    |     |
+    |  | EntryK  | ...     | ...     |                                                  |    |     |
+    |  +---------+---------+---------+                                                  |    |     |
+    |                                                                                   |    |     |
+    |                                                                                   |    |     |
+    |  +---------------+  +----------+  +-------------+                                 v    v     |
+    +->| leafnode      |  | leafnode |  | leafnode    |               +------------+------------+  |
+       +===============+  +==========+  +=============+               | Block0     | Block1     |  |
+       | OID           |  | OID      |  | OID         |               +============+============+  |
+       | Block no      |  | Block no |  | Block no    |               | Free       |            |  |
+       | Offset        |  | Offset   |  | Offset      |               +------------+            +  |
+       | Extent        |  | Extent   |  | Extent      |----------+    | Small data | Data       |  |
+       +---------------+  +----------+  +-------------+          |    +------------+            +  |
+                                                                 +-+->| Extent     |            |  |
+                                                                      +------------+------------+  |
+                                                                          |                        |
+                                                                          +------------------------+
+
+
+WAL
+---
+Each AG has a WAL.
+The datas written to the WAL are all metadta updates, free space update and data.
+Note that only data smaller than the predefined threshold needs to be written to the WAL.
+The larger data is written to the unallocated free space and its onode's extent_map is updated accordingly 
+(also on-disk extent map). We statically allocate WAL partition aside from data partition pre-configured.
+
+
+Partition
+---------
+Initially, PoseidonStore employs static allocation of partition. The number of sharded partitions
+is fixed and the size of each partition also should be configured before running cluster.
+
+
+Cache
+-----
+There are mainly two cache data structures; onode cache and block cache.
+Since PoseidonStore does not have the on-disk onode strucutre, there needs to be in-memory onode structure 
+for fast tracking of the objects.
```

Comment:
I asked above, but is the idea that you don't care about the ondisk onode representation *because* you assume you'll be able to cache all of them?